lib/excon/constants.rb

```diff
@@ -119,10 +119,10 @@ module WaitWritable; end
     :idempotent           => false,
     :instrumentor_name    => 'excon',
     :middlewares          => [
+      Excon::Middleware::Instrumentor,
```

Comment:
Makes me wonder if there should be a retry callback also (or something to that effect). The current setup somewhat limits what can be done here.

Perhaps the instrumentor could check for retries and behave differently? It should be able to figure out via comparing to retries_remaining. Though it may be better/cleaner to actually add a retry boolean or something to the data, or at least a method that did the comparison instead of leaking that out across even more places.